Zend/zend_vm_def.h

```diff
@@ -3167,24 +3167,85 @@ ZEND_VM_HANDLER(128, ZEND_INIT_DYNAMIC_CALL, ANY, CONST|TMPVAR|CV)
 
 ZEND_VM_C_LABEL(try_function_name):
 	if (OP2_TYPE != IS_CONST && EXPECTED(Z_TYPE_P(function_name) == IS_STRING)) {
-		if (Z_STRVAL_P(function_name)[0] == '\\') {
-			lcname = zend_string_alloc(Z_STRLEN_P(function_name) - 1, 0);
-			zend_str_tolower_copy(lcname->val, Z_STRVAL_P(function_name) + 1, Z_STRLEN_P(function_name) - 1);
+		const char *colon;
+		
+		if ((colon = zend_memrchr(Z_STRVAL_P(function_name), ':', Z_STRLEN_P(function_name))) != NULL &&
+			colon > Z_STRVAL_P(function_name) &&
+			*(colon-1) == ':'
+		) {
+			zend_string *mname;
+			size_t cname_length = colon - Z_STRVAL_P(function_name) - 1;
+			size_t mname_length = Z_STRLEN_P(function_name) - cname_length - (sizeof("::") - 1);
+			
+			if (!mname_length) {
+				zend_error(E_EXCEPTION | E_ERROR, "Call to undefined function %s()", Z_STRVAL_P(function_name));
+				FREE_OP2();
+				HANDLE_EXCEPTION();
+			}
+			
+			lcname = zend_string_init(Z_STRVAL_P(function_name), cname_length, 0);
+			
+			called_scope = zend_fetch_class_by_name(lcname, NULL, ZEND_FETCH_CLASS_DEFAULT | ZEND_FETCH_CLASS_EXCEPTION);
+			if (UNEXPECTED(called_scope == NULL)) {
+				zend_string_release(lcname);
```

Comment:
hmm, then okey, I thought a similar error "call to undefined" should be triggered. but fine, let keep them consistent.. thanks